openshift-kube-apiserver/admission/customresourcevalidation/node/validate_node_config.go

```diff
@@ -0,0 +1,218 @@
+	if !ok {
+		return nil, append(allErrs,
+			field.NotSupported(field.NewPath("kind"), fmt.Sprintf("%T", uncastObj), []string{"Node"}),
+			field.NotSupported(field.NewPath("apiVersion"), fmt.Sprintf("%T", uncastObj), []string{"config.openshift.io/v1"}))
+	}
+
+	return obj, nil
+}
+
+type configNodeV1 struct {
+	nodeListerFn                 func() corev1listers.NodeLister
+	waitForNodeInformerSyncedFn  func() func() bool
+	minimumKubeletVersionEnabled bool
+}
+
+func validateConfigNodeForExtremeLatencyProfile(obj, oldObj *configv1.Node) *field.Error {
+	fromProfile := oldObj.Spec.WorkerLatencyProfile
+	toProfile := obj.Spec.WorkerLatencyProfile
+
+	for _, rejectionScenario := range rejectionScenarios {
+		if fromProfile == rejectionScenario.fromProfile && toProfile == rejectionScenario.toProfile {
+			return field.Invalid(field.NewPath("spec", "workerLatencyProfile"), obj.Spec.WorkerLatencyProfile,
+				fmt.Sprintf(
+					"cannot update worker latency profile from %q to %q as extreme profile transition is unsupported, please select any other profile with supported transition such as %q",
+					oldObj.Spec.WorkerLatencyProfile,
+					obj.Spec.WorkerLatencyProfile,
+					configv1.MediumUpdateAverageReaction,
+				),
+			)
+		}
+	}
+	return nil
+}
+
+func (c *configNodeV1) ValidateCreate(_ context.Context, uncastObj runtime.Object) field.ErrorList {
+	obj, allErrs := toConfigNodeV1(uncastObj)
+	if len(allErrs) > 0 {
+		return allErrs
+	}
+
+	allErrs = append(allErrs, validation.ValidateObjectMeta(&obj.ObjectMeta, false, customresourcevalidation.RequireNameCluster, field.NewPath("metadata"))...)
+	if err := c.validateMinimumKubeletVersion(obj); err != nil {
+		allErrs = append(allErrs, err)
+	}
+
+	return allErrs
+}
+
+func (c *configNodeV1) ValidateUpdate(_ context.Context, uncastObj runtime.Object, uncastOldObj runtime.Object) field.ErrorList {
+	obj, allErrs := toConfigNodeV1(uncastObj)
+	if len(allErrs) > 0 {
+		return allErrs
+	}
+	oldObj, allErrs := toConfigNodeV1(uncastOldObj)
+	if len(allErrs) > 0 {
+		return allErrs
+	}
+
+	allErrs = append(allErrs, validation.ValidateObjectMetaUpdate(&obj.ObjectMeta, &oldObj.ObjectMeta, field.NewPath("metadata"))...)
+	if err := validateConfigNodeForExtremeLatencyProfile(obj, oldObj); err != nil {
+		allErrs = append(allErrs, err)
+	}
+	if err := c.validateMinimumKubeletVersion(obj); err != nil {
+		allErrs = append(allErrs, err)
+	}
+
+	return allErrs
+}
+func (c *configNodeV1) validateMinimumKubeletVersion(obj *configv1.Node) *field.Error {
+	if !c.minimumKubeletVersionEnabled {
+		return nil
+	}
+	fieldPath := field.NewPath("spec", "minimumKubeletVersion")
+	if !c.waitForNodeInformerSyncedFn()() {
+		return field.InternalError(fieldPath, fmt.Errorf("caches not synchronized, cannot validate minimumKubeletVersion"))
+	}
+
+	nodes, err := c.nodeListerFn().List(labels.Everything())
+	if err != nil {
+		return field.Forbidden(fieldPath, fmt.Sprintf("Getting nodes to compare minimum version %v", err.Error()))
```

Comment:
I don't think the type of the error matters much because these errors are reported as `apierrors.NewInvalid`, but maybe we could change this one to `field.NotFound(...)`